pulp_ansible/app/settings.py

```diff
@@ -22,3 +22,5 @@
 
 ANSIBLE_API_HOSTNAME = "https://" + socket.getfqdn()
 ANSIBLE_CONTENT_HOSTNAME = settings.CONTENT_ORIGIN + "/pulp/content"
+ANSIBLE_DEFAULT_DISTRIBUTION_PATH = None
+ANSIBLE_URL_NAMESPACE = ""
```

Comment:
Adding docs with some examples would help me understand this better (and we need it to merge anyway). I think put them [here](https://github.com/pulp/pulp_ansible/blob/main/docs/settings.rst).

Reply:
@mdellweg @bmbouter there’s one thing I wanted to get your feedback on. One of the problems I had to solve was to figure out how to get pulp ansible to route correctly to the galaxy_ng apis any time reverse is called. The solution I came up with is the ANSIBLE_URL_NAMESPACE setting which basically allows you to inject a router namespace into the reverse lookup which tells pulp ansible to use the URL routes configured under the galaxy_ng app instead (that’s what’s happening here: https://github.com/pulp/pulp_ansible/pull/798/files#diff-d74e2c9333d6150adbd3c142482e4ba75697f468d983b35e696ab7e4fcfe8ceaR64)

There’s another potential solution to this problem and that’s to just set the `GALAXY_API_ROOT` in pulp ansible to the same as the API root in galaxy ng. I tested this and it seems to work, but the problem I ran into is that it will also include v1 and v2 of the API and I couldn’t figure out how to solve that without making breaking changes to the `GALAXY_API_ROOT` setting. I sort of like this approach better because then there’s no separation between the pulp and galaxy APIs, but it does make it a little bit harder to declare when we don’t want a route from pulp ansible like the v1 and v2 apis and eventually the configuration endpoint, which will have to be different in galaxy ng.

I was wondering what you think about these two approaches?

Reply:
I'm not sure to see all aspects of the challenge here. First we need to make sure, that pulp_ansible has a consistent api on it's own. So with setting `GALAXY_API_ROOT` you can basically move the location of pulp_ansible's galaxy-api around? What if we introduce settings to move the versions of the api independently (`GALAXY_APIV1_ROOT`, ...)? Or even provide a setting that describes which versions of the api we want to expose at all?